app/lib/json_web_token.rb

```diff
@@ -6,6 +6,8 @@ def self.encode(payload, expiration = nil)
     payload['exp'] = expiration.to_i.hours.from_now.to_i
 
     JWT.encode(payload, Rails.application.secrets.jwt_secret)
+  rescue => e
+    raise "Failed to encode JsonWebToken due to: #{e}"
```

Comment:
@rickr - Is this more inline with what you were looking for?  This will raise if `jwt_secret` is missing, as well as any other issues in the `encode` method.

Or did you want to `raise` **only if** `jwt_secret` is `nil`?  i.e.

```ruby
  def self.encode(payload, expiration = nil)
    expiration ||= Rails.application.secrets.jwt_expiration_hours

    payload = payload.dup
    payload['exp'] = expiration.to_i.hours.from_now.to_i

    raise "Rails.application.secrets.jwt_secret is nil. Discovered in JsonWebToken.encode." if Rails.application.secrets.jwt_secret.nil?

    JWT.encode(payload, Rails.application.secrets.jwt_secret)
  end
```